src/GameEngine/GameEngine.cpp

```diff
@@ -1,67 +1,148 @@
 #include "GameEngine.h"
 
-//-------------------------------------------Class State---------------------------------------------------------------
+/************************************************************ State **************************************************************/
+/**
+ * Default Constructor
+ */
+    State::State(){}
+/**
+ * Destructor
+ */
+    State::~State(){}
+/**
+ * Constructor with an argument(s)
+ */
+    State::State(string name){ 
+        stateName = name;
+    }
+/**
+ * Assignment Operator
+ */
+    State& State::operator=(const State& other){
+        this->stateName = other.stateName;
+        return *this;
+    }
+/**
+ * Copy Constructor
+ */
+    State::State(const State& other) {
+        this->stateName = other.stateName;
+    }
+/**
+ * overide Stream insertion operator
+ */
+    ostream& operator<<(ostream& out, State* s){
+        cout << "You are in " << s->stateName << "\n";
```

Comment:
you wanna do ` out << "You are in " << s->stateName\n";`
since you wanna utilize the stream operator you are getting back :)

and no need to create a new stream using `<<` u can just concatenate the "\n"